### ISSUE TITLE
False positive with return guard

### ISSUE THREAD

**Issue by johndoknjas** [pylint-dev/pylint#9464]
### Bug description

```python
import random
from typing import Optional

var: Optional[int] = None

def maybe_modify() -> None:
    global var
    if random.randint(0, 1) == 0:
        var = 1

def test1() -> None:
    global var
    maybe_modify()
    if var is None:
        return
    print(-var)

def test2() -> None:
    global var
    maybe_modify()
    if var is None:
        pass
    else:
        print(-var)

def main():
    test2() # pylint correctly has no issues with test2
    test1() # but has a problem with test1

if __name__ == '__main__':
    main()
```

### Command used

```shell
pylint main.py
```

### Pylint output

```python
************* Module main
main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main.py:6:0: C0116: Missing function or method docstring (missing-function-docstring)
main.py:7:4: W0603: Using the global statement (global-statement)
main.py:11:0: C0116: Missing function or method docstring (missing-function-docstring)
main.py:12:4: W0602: Using global for 'var' but no assignment is done (global-variable-not-assigned)
main.py:16:10: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
main.py:18:0: C0116: Missing function or method docstring (missing-function-docstring)
main.py:19:4: W0602: Using global for 'var' but no assignment is done (global-variable-not-assigned)
main.py:26:0: C0116: Missing function or method docstring (missing-function-docstring)

------------------------------------------------------------------
Your code has been rated at 4.58/10 (previous run: 4.58/10, +0.00)
```

### Expected behavior

The expected behaviour would be the same output as above, except with the E1130 line omitted.

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
```

### OS / Environment

Windows 11, Windows PowerShell
